### ISSUE TITLE
Allow whitespace in alias names

### ISSUE THREAD

**Issue by hyiltiz** [nushell/nushell#8103]
### Related problem

Allowing whitespace ` ` in defined alias names would provide a consistent interface just like function definitions. E.g. currently in `0.75.1`, after creating alias like the following does not actually create the aliases (`go prod` says no command found):
```
alias 'go prod' = 'a complex command that uses overlay hence can't be wrapped inside a def function block'
alias 'go test = 'a complex command that uses overlay hence can't be wrapped inside a def function block'
```

### Describe the solution you'd like

`go prod` simply expands to the alias string it is defined to, interpolates it and feeds it into the CLI and evaluates it.

### Describe alternatives you've considered

Wrapping inside a `def 'go prod' []{some command that is used to export overlay}` fails as the exported overlay is scoped within the function, and does not modify the CLI environment.

### Additional context and details

An example usage would be to trigger virtual environments (like python virtual environments) by simple commands like `go py proj1` that would overlay a corresponding `activate.nu` file provided by `python3-virtualenv` package.

**Comment by amtoine** [comment]
@hyiltiz 

i was confused because i've already defined multi-words aliases, see , so i tested this on f3ee8b50e3da9548fe39cfb104a20d6da10a6458 and here is what i found :yum: 

the problem appears to be the single quotes :open_mouth: 
=> in different shells each time
```bash
>_ nu
>_ alias "go prod" = 'this is `go prod`'
>_ go prod
this is `go prod`
>_ exit
```
```bash
>_ nu
>_ alias "go prod" = echo 'this is `go prod`'
>_ go prod
this is `go prod`
>_ exit
```
```bash
>_ nu
>_ alias 'go prod' = 'this is `go prod`'
>_ go prod
go prod: unknown command
Run 'go help' for usage.
>_ exit
```
```bash
>_ nu
>_ alias 'go prod' = echo 'this is `go prod`'
>_ go prod
go prod: unknown command
Run 'go help' for usage.
>_ exit
```

**Comment by amtoine** [comment]
**TL;DR**:
- `alias "go prod"` appears to work either with a literal of with a command
- `alias 'go prod'` does not appear to work at all

**Comment by amtoine** [comment]
> Hmm, that seems quite unexpected, no? In `nu`, both `'` and `"` constructs the same `string` type. It is quite surprising for them to behave differently.
>
depending on the exact place where you use them, these three can behave quite differently
### example with a simple dummy regex
- interpreted as a command
```bash
>_ `\d`
Error: nu::shell::external_command (link)

  × External command failed
   ╭─[entry #13:1:1]
 1 │ `\d`
   · ──┬─
   ·   ╰── did you mean 'cd'?
   ╰────
  help: No such file or directory (os error 2)
```
- not interpreted at all, giving exact regex
```bash
>_ '\d'
\d
```
- `nushell` trying to interpret the escaped character
```bash
>_ "\d"
Error:
  × Invalid literal
   ╭─[entry #15:1:1]
 1 │ "\d"
   ·  ─┬─
   ·   ╰── unrecognized escape after '\' in string
   ╰────
```

> While we are on it, I found that constructing the alias name with backtick `` ` `` (wanna know how to write backtick in Markdown code chunk? :-P) also doesn't work.
> 
> It only works with `"`; but it should really work across all string constructors.
>
not sure what the behaviour of this should be regarding `alias`es :thinking:

**Comment by hyiltiz** [comment]
For reference, here [1] is the current description for `'` `"` `` ` `` 
[1] https://www.nushell.sh/book/working_with_strings.html

I think, all of these constructors should behave consistently, with the only difference being what's documented in [1] (TL;DR their content cannot contain the constructor itself without escape magic).

**Comment by hyiltiz** [comment]
It looks like the recent changes to the alias system broke the ability to define multi-word aliases, so :
```
> alias "go prod" = 'this is `go prod`'
> go prod # fails
> alias "go prod" = echo 'this is `go prod`'
> go prod # works
```

I was using an alias `alias "go venv" = overlay use .venv/bin/activate.nu` along with this [1] activate.nu, so I'd create a `venv`, copy `activate.nu` into `./.venv/bin/`, and from then on, I could just do `go venv`; now this workflow is broken. I had similar set up for things like setting JAVA_HOME etc. depending on various projects, with `go projA` that would set up the "environment" for the project.

I think it would be much better if there were a standardized way of setting up project/folder specific environments (for projects like Python, Haskell, Java, JavaScript etc.) This would probably be based on`export-env` functionality, similar to [2]. I could never get [2] to work either (hence monkey-patching it and removing its "auto-trigger-on-cd" behavior with [1].

[1] https://paste.tomsmeding.com/RySNNAV4
[2] https://github.com/nushell/nu_scripts/tree/main/virtual_environments/auto-venv
